src/coap_gnutls.c

```diff
@@ -540,42 +567,53 @@ psk_client_callback(gnutls_session_t g_session,
   return 0;
 }
 
+typedef struct {
+  gnutls_certificate_type_t certificate_type;
+  char *san_or_cn;
+  const gnutls_datum_t *cert_list;
+  unsigned int cert_list_size;
+} coap_gnutls_certificate_info_t;
+
 /*
- * return +ve  SAN or CN derived from certificate
- *        NULL failed
+ * return Type of certificate and SAN or CN if appropriate derived from
+ *        certificate. COAP_GNUTLS_CRT_ERROR if failure.
```

Comment:
In case of error, this function returns GNUTLS_CRT_UNKNOWN

Reply:
Oops - corrected.